frontend/src/routes/Infrastructure/Clusters/ClusterSets/ClusterSetDetails/ClusterSetInstallSubmariner/InstallSubmarinerForm.tsx

```diff
@@ -289,7 +289,11 @@ export function InstallSubmarinerForm(props: { availableClusters: Cluster[] }) {
                 }
 
                 // configure secret if one doesn't exist
-                if (cluster.provider !== Provider.vmware && providerSecretMap[cluster.displayName!] === null) {
+                if (
```

Comment:
```suggestion
                               // Create credential secret if one doesn't exist
                const secret: Secret = {
                    apiVersion: SecretApiVersion,
                    kind: SecretKind,
                    metadata: {
                        name: `${cluster.name}-${cluster.provider}-creds`,
                        namespace: cluster.namespace,
                    },
                    stringData: {},
                    type: 'Opaque',
                }

                // configure secret if one doesn't exist
                if (cluster.provider !== Provider.vmware && providerSecretMap[cluster.displayName!] === null) {
                if (
                    cluster.provider !== Provider.vmware &&
                    providerSecretMap[cluster.displayName!] === null &&
                    !cluster.isManaged!
                ) {
                    if (cluster.provider === Provider.aws) {
                        secret.stringData!['aws_access_key_id'] = awsAccessKeyIDs[cluster.displayName!]! || ''
                        secret.stringData!['aws_secret_access_key'] = awsSecretAccessKeyIDs[cluster.displayName!]! || ''
                    } else if (cluster.provider === Provider.gcp) {
                        secret.stringData!['osServiceAccount.json'] = gcServiceAccountKeys[cluster.displayName!]! || ''
                    } else if (cluster.provider === Provider.azure) {
                        const clientid = clientIds[cluster.displayName!]! || ''
                        const clientSecret = clientSecrets[cluster.displayName!]! || ''
                        const tenantid = tenantIds[cluster.displayName!]! || ''
                        const subscriptionid = subscriptionIds[cluster.displayName!]! || ''
                        secret.stringData!['osServicePrincipal.json'] = JSON.stringify({
                            clientid,
                            clientSecret,
                            tenantid,
                            subscriptionid,
                        })
                        secret.stringData!['baseDomainResourceGroupName'] =
                            baseDomainResourceGroupNames[cluster.displayName!]! || ''
                    }
                    resources.push(secret)
                }
                const submarinerConfig: SubmarinerConfig = {
                    apiVersion: SubmarinerConfigApiVersion,
                    kind: SubmarinerConfigKind,
                    metadata: {
                        name: 'submariner',
                        namespace: cluster?.namespace!,
                    },
                    spec: {
                        gatewayConfig: {
                            gateways: gateways[cluster.displayName!] || submarinerConfigDefault.gateways,
                        },
                        IPSecNATTPort: nattPorts[cluster.displayName!] ?? submarinerConfigDefault.nattPort,
                        NATTEnable: nattEnables[cluster.displayName!] ?? submarinerConfigDefault.nattEnable,
                        cableDriver: cableDrivers[cluster.displayName!] ?? submarinerConfigDefault.cableDriver,
                    },
                }
                if (cluster.provider !== Provider.vmware) {
                    // use existing secret name
                    if (providerSecretMap[cluster.displayName!]) {
                        submarinerConfig.spec.credentialsSecret = {
                            name: providerSecretMap[cluster.displayName!]!,
                        }
                    } else {
                        // use secret name that will be created
                        submarinerConfig.spec.credentialsSecret = {
                            name: secret.metadata.name!,
                        }
                    }
                }
                // configure instance type if AWS
                if (cluster.provider === Provider.aws) {
                    submarinerConfig.spec.gatewayConfig!.aws = {
                        instanceType: awsInstanceTypes[cluster.displayName!] ?? submarinerConfigDefault.awsInstanceType,
                    }
                } else if (cluster.provider === Provider.azure) {
                    submarinerConfig.spec.gatewayConfig!.azure = {
                        instanceType:
                            azInstanceTypes[cluster.displayName!] ?? submarinerConfigDefault.azureInstanceType,
                    }
                }
                resources.push(submarinerConfig)
```
                const submarinerConfig: SubmarinerConfig = {
                    apiVersion: SubmarinerConfigApiVersion,
                    kind: SubmarinerConfigKind,
                    metadata: {
                        name: 'submariner',
                        namespace: cluster?.namespace!,
                    },
                    spec: {
                        gatewayConfig: {
                            gateways: gateways[cluster.displayName!] || submarinerConfigDefault.gateways,
                        },
                        IPSecNATTPort: nattPorts[cluster.displayName!] ?? submarinerConfigDefault.nattPort,
                        NATTEnable: nattEnables[cluster.displayName!] ?? submarinerConfigDefault.nattEnable,
                        cableDriver: cableDrivers[cluster.displayName!] ?? submarinerConfigDefault.cableDriver,
                    },
                }

                if (!cluster.distribution?.isManagedOpenShift) {
                    // Create credential secret if one doesn't exist
                    const secret: Secret = {
                        apiVersion: SecretApiVersion,
                        kind: SecretKind,
                        metadata: {
                            name: `${cluster.name}-${cluster.provider}-creds`,
                            namespace: cluster.namespace,
                        },
                        stringData: {},
                        type: 'Opaque',
                    }

                    // configure secret if one doesn't exist
                    if (
                        cluster.provider !== Provider.vmware &&
                        providerSecretMap[cluster.displayName!] === null
                    ) {
                        if (cluster.provider === Provider.aws) {
                            secret.stringData!['aws_access_key_id'] = awsAccessKeyIDs[cluster.displayName!]! || ''
                            secret.stringData!['aws_secret_access_key'] = awsSecretAccessKeyIDs[cluster.displayName!]! || ''
                        } else if (cluster.provider === Provider.gcp) {
                            secret.stringData!['osServiceAccount.json'] = gcServiceAccountKeys[cluster.displayName!]! || ''
                        } else if (cluster.provider === Provider.azure) {
                            const clientid = clientIds[cluster.displayName!]! || ''
                            const clientSecret = clientSecrets[cluster.displayName!]! || ''
                            const tenantid = tenantIds[cluster.displayName!]! || ''
                            const subscriptionid = subscriptionIds[cluster.displayName!]! || ''
                            secret.stringData!['osServicePrincipal.json'] = JSON.stringify({
                                clientid,
                                clientSecret,
                                tenantid,
                                subscriptionid,
                            })
                            secret.stringData!['baseDomainResourceGroupName'] =
                                baseDomainResourceGroupNames[cluster.displayName!]! || ''
                        }

                        resources.push(secret)
                    }

                    if (cluster.provider !== Provider.vmware) {
                        // use existing secret name
                        if (providerSecretMap[cluster.displayName!]) {
                            submarinerConfig.spec.credentialsSecret = {
                                name: providerSecretMap[cluster.displayName!]!,
                            }
                        } else {
                            // use secret name that will be created
                            submarinerConfig.spec.credentialsSecret = {
                                name: secret.metadata.name!,
                            }
                        }
                    }

                    // configure instance type if AWS
                    if (cluster.provider === Provider.aws) {
                        submarinerConfig.spec.gatewayConfig!.aws = {
                            instanceType: awsInstanceTypes[cluster.displayName!] ?? submarinerConfigDefault.awsInstanceType,
                        }
                    } else if (cluster.provider === Provider.azure) {
                        submarinerConfig.spec.gatewayConfig!.azure = {
                            instanceType:
                                azInstanceTypes[cluster.displayName!] ?? submarinerConfigDefault.azureInstanceType,
                        }
                    }
                }
                resources.push(submarinerConfig)